### PR TITLE
feat(apigateway): Service and checks for APIGW v1 and v2

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Context 

Add checks, services and tests for ApiGateway version 1 & 2 since we need to migrate them to the new version 3.

### Description

Add checks, services and tests for ApiGateway version 1 & 2 

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jfagoagas** [comment]
Test

File: providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        ):
+            # Test Check
+            from providers.aws.services.apigateway.apigateway_authorizers_enabled.apigateway_authorizers_enabled import (
+                apigateway_authorizers_enabled,
+            )
+
+            check = apigateway_authorizers_enabled()
+            result = check.execute()
+
+            assert len(result) == 0
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py:30]: Add assertion to other fields of the `Check_Report` like `resource_id` or `status_extended`

File: providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        ):
+            # Test Check
+            from providers.aws.services.apigateway.apigateway_authorizers_enabled.apigateway_authorizers_enabled import (
+                apigateway_authorizers_enabled,
+            )
+
+            check = apigateway_authorizers_enabled()
+            result = check.execute()
+
+            assert result[0].status == "PASS"
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py:80]: Same here

File: providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        ):
+            # Test Check
+            from providers.aws.services.apigateway.apigateway_authorizers_enabled.apigateway_authorizers_enabled import (
+                apigateway_authorizers_enabled,
+            )
+
+            check = apigateway_authorizers_enabled()
+            result = check.execute()
+
+            assert result[0].status == "FAIL"
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py:113]: Same as above

File: providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        ):
+            # Test Check
+            from providers.aws.services.apigateway.apigateway_authorizers_enabled.apigateway_authorizers_enabled import (
+                apigateway_authorizers_enabled,
+            )
+
+            check = apigateway_authorizers_enabled()
+            result = check.execute()
+
+            assert len(result) == 0
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py]: Same as above

File: providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        ):
+            # Test Check
+            from providers.aws.services.apigateway.apigateway_client_certificate_enabled.apigateway_client_certificate_enabled import (
+                apigateway_client_certificate_enabled,
+            )
+
+            check = apigateway_client_certificate_enabled()
+            result = check.execute()
+
+            assert len(result) == 0
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py:37]: Same as other tests

File: providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+            from providers.aws.services.apigateway.apigateway_client_certificate_enabled.apigateway_client_certificate_enabled import (
+                apigateway_client_certificate_enabled,
+            )
+
+            check = apigateway_client_certificate_enabled()
+            result = check.execute()
+
+            assert result[0].status == "FAIL"
+
+    # @mock_apigateway
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py]: Delete this comment if it is not needed

File: providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        ):
+            # Test Check
+            from providers.aws.services.apigateway.apigateway_client_certificate_enabled.apigateway_client_certificate_enabled import (
+                apigateway_client_certificate_enabled,
+            )
+
+            check = apigateway_client_certificate_enabled()
+            result = check.execute()
+
+            assert result[0].status == "FAIL"
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py:92]: Same as other tests

File: providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+        ):
+            # Test Check
+            from providers.aws.services.apigateway.apigateway_client_certificate_enabled.apigateway_client_certificate_enabled import (
+                apigateway_client_certificate_enabled,
+            )
+
+            check = apigateway_client_certificate_enabled()
+            result = check.execute()
+
+            assert len(result) == 0
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py]: Same as other tests

File: providers/aws/services/apigateway/apigateway_endpoint_public/apigateway_endpoint_public_test.py [hunk 9]
```diff
@@ -0,0 +1,111 @@
+from unittest import mock
+
+from boto3 import client
+from moto import mock_apigateway
+
+AWS_REGION = "us-east-1"
+
+
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_endpoint_public/apigateway_endpoint_public_test.py:8]: Check all the assertions

File: providers/aws/services/apigateway/apigateway_endpoint_public/apigateway_endpoint_public_test.py [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+                apigateway_endpoint_public,
+            )
+
+            check = apigateway_endpoint_public()
+            result = check.execute()
+
+            assert result[0].status == "PASS"
+
+    @mock_apigateway
+    def test_apigateway_one_prublic_rest_api(self):
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_endpoint_public/apigateway_endpoint_public_test.py]:
```suggestion
    def test_apigateway_one_public_rest_api(self):
```

File: providers/aws/services/apigateway/apigateway_logging_enabled/apigateway_logging_enabled_test.py [hunk 11]
```diff
@@ -0,0 +1,166 @@
+from unittest import mock
+
+from boto3 import client
+from moto import mock_apigateway
+
+AWS_REGION = "us-east-1"
+
+
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_logging_enabled/apigateway_logging_enabled_test.py:8]: Check all the assertions

File: providers/aws/services/apigateway/apigateway_waf_acl_attached/apigateway_waf_acl_attached_test.py [hunk 12]
```diff
@@ -0,0 +1,172 @@
+from unittest import mock
+
+from boto3 import client
+from moto import mock_apigateway, mock_wafv2
+
+AWS_REGION = "us-east-1"
+
+
```
Comment (n4ch04) [inline providers/aws/services/apigateway/apigateway_waf_acl_attached/apigateway_waf_acl_attached_test.py:8]: Check all the assertions

File: providers/aws/services/apigatewayv2/apigatewayv2_access_logging_enabled/apigatewayv2_access_logging_enabled_test.py [hunk 13]
```diff
@@ -0,0 +1,102 @@
+from unittest import mock
+
+import botocore
```
Comment (n4ch04) [inline providers/aws/services/apigatewayv2/apigatewayv2_access_logging_enabled/apigatewayv2_access_logging_enabled_test.py:3]: Check all the assertions

File: providers/aws/services/apigatewayv2/apigatewayv2_authorizers_enabled/apigatewayv2_authorizers_enabled_test.py [hunk 14]
```diff
@@ -0,0 +1,108 @@
+from unittest import mock
+
+import botocore
```
Comment (n4ch04) [inline providers/aws/services/apigatewayv2/apigatewayv2_authorizers_enabled/apigatewayv2_authorizers_enabled_test.py:3]: Check all the assertions

**Review by n4ch04 — DISMISSED** [review]

File: providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py (hunk 1 above)
Comment (MrCloudSec) [reply providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py:30]: Done

File: providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py (hunk 2 above)
Comment (MrCloudSec) [reply providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py:80]: Done

File: providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py (hunk 3 above)
Comment (MrCloudSec) [reply providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py:113]: Done

File: providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py (hunk 4 above)
Comment (MrCloudSec) [reply providers/aws/services/apigateway/apigateway_authorizers_enabled/apigateway_authorizers_enabled_test.py]: Done

File: providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py (hunk 5 above)
Comment (MrCloudSec) [reply providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py:37]: Done

File: providers/aws/services/apigateway/apigateway_logging_enabled/apigateway_logging_enabled_test.py (hunk 11 above)
Comment (MrCloudSec) [reply providers/aws/services/apigateway/apigateway_logging_enabled/apigateway_logging_enabled_test.py:8]: Done

File: providers/aws/services/apigateway/apigateway_waf_acl_attached/apigateway_waf_acl_attached_test.py (hunk 12 above)
Comment (MrCloudSec) [reply providers/aws/services/apigateway/apigateway_waf_acl_attached/apigateway_waf_acl_attached_test.py:8]: Done

File: providers/aws/services/apigatewayv2/apigatewayv2_access_logging_enabled/apigatewayv2_access_logging_enabled_test.py (hunk 13 above)
Comment (MrCloudSec) [reply providers/aws/services/apigatewayv2/apigatewayv2_access_logging_enabled/apigatewayv2_access_logging_enabled_test.py:3]: Done

File: providers/aws/services/apigatewayv2/apigatewayv2_authorizers_enabled/apigatewayv2_authorizers_enabled_test.py (hunk 14 above)
Comment (MrCloudSec) [reply providers/aws/services/apigatewayv2/apigatewayv2_authorizers_enabled/apigatewayv2_authorizers_enabled_test.py:3]: Done

File: providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py (hunk 7 above)
Comment (jfagoagas) [reply providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py:92]: Done

File: providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py (hunk 6 above)
Comment (jfagoagas) [reply providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py]: Done

Comment (jfagoagas) [reply providers/aws/services/apigateway/apigateway_client_certificate_enabled/apigateway_client_certificate_enabled_test.py]: Pending mocked test

**Review by jfagoagas — CHANGES_REQUESTED** [review]

**Review by jfagoagas — APPROVED** [review]
Let's go 🚀

**Pull request merged**